### PR TITLE
doc(examples): improve example documentation

### PR DESCRIPTION
**Pull request by vados-cosmonic**
## Feature or Problem

Improve documentation around examples -- how to run them and context on what they're meant to do/display.

### CODE REVIEW

File: examples/README.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+**Want to implement an interface for reading and writing key-value data that speaks to a real database? That's a wasmCloud provider.**
+
+wasmCloud capability providers run as child processes on the wasmCloud host, and are mostly unrestricted, meaning you don't have to wait for WebAssembly ecosystem support for your code -- whatever advanced functionality (connecting to databases, AI/ML, raw hardware access, proprietary software, etc) you can build into a regular binary today can exposed for WebAssembly components running on your lattice to consume.
+
+You can get started quickly by trying out of our example projects:
+
+| Language     | Folder                                                                                 | OCI Artifact |
+|--------------|----------------------------------------------------------------------------------------|--------------|
+| [Rust][rust] | [`./rust/components/http-keyvalue-counter`](./golang/components/http-keyvalue-counter) |              |
```
Comment (brooksmtownsend) [inline examples/README.md:66]: This should be the messaging provider right?

File: examples/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+> [!WARNING]
+> Most wasmCloud example providers do *not* have officially maintained OCI artifacts.
+>
+> Want to use a proivder from your own registry? Build a Proivder ARchive (PAR) file with `wash par` and `wash push` it to your registry of choice.
```
Comment (brooksmtownsend) [inline examples/README.md:71]:
```suggestion
> Want to use a provider from your own registry? Build a Provider Archive (PAR) file with `wash build` and `wash push` it to your registry of choice.
```

**Review by brooksmtownsend — COMMENTED** [review]
Largely looks wonderful just a couple things to address 🚀

File: examples/README.md (hunk 2 above)
Comment (vados-cosmonic) [reply examples/README.md]: Thanks for this -- fixed the typos here

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**